base/src/sgpp/base/datatypes/DataMatrix.cpp

```diff
@@ -183,26 +187,26 @@ size_t DataMatrix::appendRow() {
   this->insert(this->end(), this->ncols, 0.0);
 
   size_t x = nrows;
-  ++this->nrows;
+  this->nrows++;
 
   return x;
 }
 
 void DataMatrix::transpose() {
   if (this->nrows == this->ncols) {
-    for (size_t i = 1; i < this->nrows; ++i) {
-      for (size_t j = 0; j < i; ++j) {
+    for (size_t i = 1; i < this->nrows; i++) {
+      for (size_t j = 0; j < i; j++) {
```

Comment:
is there an advantage in postfix increment here?
assuming it gets called quite often, prefix should perform a bit better.

Reply:
Prefix/postfix are actually the same for integer types. It only matters for classes with overloaded operators (usually iterators).

Reason behind these changes is consistency: Making the diff between DP and SP as small as possible (DP used prefix and SP used postfix), which will make it easier to keep them synchronized.

It don't care whether we use prefix or postfix in these classes, but it should be consistent.